central/vulnerabilityrequest/datastore/datastore_impl_test.go

```diff
@@ -188,48 +188,56 @@ func (s *VulnRequestDataStoreTestSuite) TestUpdateStatus() {
-			expectedReqObj:     pendingReq.GetReq(),
-			expectedUpdatedObj: nil,
+			name:                            "Pending request should be approved",
+			existingReq:                     pendingReq.Clone(),
+			status:                          storage.RequestStatus_APPROVED,
+			expectedStatus:                  storage.RequestStatus_APPROVED,
+			expectedReqObj:                  pendingReq.GetReq(),
+			expectedUpdatedObj:              nil,
+			skipIfUnifiedCVEDeferralEnabled: false,
 		},
 		{
-			name:               "Pending request should be denied",
-			existingReq:        pendingReq.Clone(),
-			status:             storage.RequestStatus_DENIED,
-			expectedStatus:     storage.RequestStatus_DENIED,
-			expectedReqObj:     pendingReq.GetReq(),
-			expectedUpdatedObj: nil,
+			name:                            "Pending request should be denied",
+			existingReq:                     pendingReq.Clone(),
+			status:                          storage.RequestStatus_DENIED,
+			expectedStatus:                  storage.RequestStatus_DENIED,
+			expectedReqObj:                  pendingReq.GetReq(),
+			expectedUpdatedObj:              nil,
+			skipIfUnifiedCVEDeferralEnabled: false,
 		},
 		{
-			name:               "Request pending update that was denied should go back to old approved state",
-			existingReq:        pendingUpdateReq.Clone(),
-			status:             storage.RequestStatus_DENIED,
-			expectedStatus:     storage.RequestStatus_APPROVED,
-			expectedReqObj:     pendingUpdateReq.GetReq(),
-			expectedUpdatedObj: nil,
+			name:                            "Request pending update that was denied should go back to old approved state",
+			existingReq:                     pendingUpdateReq.Clone(),
+			status:                          storage.RequestStatus_DENIED,
+			expectedStatus:                  storage.RequestStatus_APPROVED,
+			expectedReqObj:                  pendingUpdateReq.GetReq(),
+			expectedUpdatedObj:              nil,
+			skipIfUnifiedCVEDeferralEnabled: false,
 		},
 		{
-			name:               "Deferral request pending update that was approved should have req updated",
-			existingReq:        pendingUpdateReq.Clone(),
-			status:             storage.RequestStatus_APPROVED,
-			expectedStatus:     storage.RequestStatus_APPROVED,
-			expectedReqObj:     &storage.VulnerabilityRequest_DeferralReq{DeferralReq: pendingUpdateReq.GetUpdatedDeferralReq()},
-			expectedUpdatedObj: nil,
+			name:                            "Deferral request pending update that was approved should have req updated",
+			existingReq:                     pendingUpdateReq.Clone(),
+			status:                          storage.RequestStatus_APPROVED,
+			expectedStatus:                  storage.RequestStatus_APPROVED,
+			expectedReqObj:                  &storage.VulnerabilityRequest_DeferralReq{DeferralReq: pendingUpdateReq.GetUpdatedDeferralReq()},
+			expectedUpdatedObj:              nil,
+			skipIfUnifiedCVEDeferralEnabled: true,
```

Comment:
Why is the test case skipped when unified deferral is enabled?

Reply:
This test creates a v1 deferral update and add the updates to a `UpdatedDeferralReq` field. When `UnifiedCVEDeferral` is enabled the implementation looks at `DeferralUpdate` or `FpUpdate` fields. So it won't apply the above update. That is why it is disabled when the FF is on. I added another test for when the FF is on.

Reply:
I guess what I'm really hinting at is that `pendingUpdateReq` is always in v1 format. For the first three test cases, you do not need to skip because the update is never applied. 
You could replicate these test cases in the newly added test for the unified deferral feature with the _correct_ `pendingUpdateReq` structure.

Reply:
Yes `pendingUpdateReq` is always in v1 format, that is why it doesn't take the below path in `applyUpdateRequest` when UnifiedCVEDeferral FF is enabled
```
    if !features.UnifiedCVEDeferral.Enabled() {
		switch r := req.GetUpdatedReq().(type) {
		case *storage.VulnerabilityRequest_UpdatedDeferralReq:
			req.Req = &storage.VulnerabilityRequest_DeferralReq{DeferralReq: r.UpdatedDeferralReq}
			// Only deferral requests can be updated for VM 1.0 exceptions
		}
		return
	}
```
So this test will fail if the feature flag is turned on. This is probably what is confusing. I can do what you suggested add all the cases in this test to `TestUpdateStatusAppliesRequestedUpdates` and also skip this whole test when FF is enabled.

Reply:
Done